addons/website_sale_slides/models/slide_channel.py

```diff
@@ -47,15 +47,14 @@ def create(self, vals_list):
 
     def write(self, vals):
         res = super(Channel, self).write(vals)
-        if 'is_published' in vals:
+        if vals.get('is_published'):
             self.filtered(lambda channel: channel.enroll == 'payment')._synchronize_product_publish()
         return res
 
     def _synchronize_product_publish(self):
         if not self:
             return
         self.filtered(lambda channel: channel.is_published and not channel.product_id.is_published).sudo().product_id.write({'is_published': True})
-        self.filtered(lambda channel: not channel.is_published and channel.product_id.is_published).sudo().product_id.write({'is_published': False})
```

Comment:
We could do something smart and unpublish the product if it is not used by any other published course?
Maybe something to double-check with Luc.

Also, sounds like a great candidate for some unit testing ^^